src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
```suggestion
public interface MachineProducer<T extends Machine> {
```
краще зробити тут прив'язку до машин щоб не можна було зробити MachineProducer<String>